### ISSUE TITLE
Modal: resizing/transition issue on every current browser

### ISSUE THREAD

**Issue by ghost** [Semantic-Org/Semantic-UI-React#2221]
Hi,

we found a bug while using the semantic **react** dialog component.

[--> Animiated Gif to show the effect <--](https://ibb.co/d2cK6R)

We can repoduce this error on
https://react.semantic-ui.com/modules/modal#modal-example-dimmer
but not with the jquery version.
So it seems a css problem base on adding / removing the scrolling class by react.

The main problem is, thats a edge case failure, that can only be repoduced if the modal dialog / the body tag will add / remove the scrolling css class and also the height and width are so chosen that the dialog doesnt fit the browser windows by missing some pixels in height and width.

For own dialogs with larger content this happens on normal sized pages, so many users of our site get this problem, when they have a special resolution.

### Steps

Open the default modal dialog dimmer example with

chrome with a html / body size of width: 345px / height: 583px
firefox with a html / body size of width: 337px / height: 574px
EDGE with a html / body size of width: 332px / height: 576px

We cant repoduce the behavior with the device emulation mode on the chrome dev tools.

### Expected Result

Stable Layout

### Actual Result

The modal dialog is switching between scolling and not scrolling.
We think the recalculation by react adds the scrolling class to the body and also to the modal dialog and remove this again and add this again and so on.

Its toggling between (on the chrome example)
```
<body class="dimmable dimmed">
...
<div class="ui modal transition visible active" style="margin-top: -289px;">
```

and

```
<body class="dimmable dimmed scrolling">
...
<div class="ui scrolling modal transition visible active" style="margin-top: -294px;">
```
Also seen on the react component

`<div class="ui modal transition visible active" style="margin-top: -289px;">`

`<div class="ui scrolling modal transition visible active" style="margin-top: -294px;">`

**Comment by layershifter** [comment]
Looks like duplicate of #1867.

**Comment by levithomason** [comment]
Thanks for the investigation.  SUIR calculates a few properties for developer convenience.  In this case, a Modal that extends taller than the screen should become scrollable.  Modals that fit within the screen height do not need to be scrollable.  This is always true and not applying the class in these cases would produce a "broken" UI for the user.

That said, the issue is that the Modal margin was not taken into account in the calculation.  I've fixed this locally, but it needs to be cleaned up for a PR.

A fix would be to measure the height + border + margin of the modal without the `scrolling` className and if that fits the viewport, then the Modal is not scrolling.

Currently, the measurement doesn't differentiate between scrolling or non-scrolling modals.  It also doesn't consider the border and margin, it only considers the height.  The height changes when going from non-scrolling to scrolling, due to the inclusion of outer margin when scrolling.

**Comment by zurez** [comment]
@levithomason  Any ETA on fix?

**Comment by toolaugh** [comment]
im too ,help me please

**Comment by brianespinosa** [comment]
@zurez @toolaugh PRs are welcome for any issues tagged as `bug` and `help wanted`.

**Comment by levithomason** [comment]
No ETA.  Please use SO and Gitter for help requests.  GitHub is reserved for performing work.

**Comment by Glathrop** [comment]
`className="scrolling"` is a workaround for this.

**Comment by Tydax** [comment]
@Glathrop Just discovered that, thanks for this workaround.
You can also change the `margin-bottom` property to `auto` of the `Modal` if you want to prevent the modal from taking the whole height if it doesn't need to.

**Comment by layershifter** [comment]
Can you check please the latest release with SUI 2.3?

**Comment by bavicj** [comment]
I have the same problem since I updated to versions 0.81.x. With 0.80.x everything was OK.
[GIF example](https://ibb.co/kQ1FhJ)

**Comment by layershifter** [comment]
Please add a minimal repro case on codesandbox.

**Comment by grean** [comment]
I personnaly fix this by adding those two lines on my modal component. The height value depending of your needs, for me it was 300..
```
<Modal
   ...
   className="scrolling"
   style={{maxHeight:'300px'}}
>...</Modal>
```

**Comment by champeleon** [comment]
Adding className='scrolling' fixes the flickering issue but I can't see the scrolling bar anymore and there's no scrolling functionality. Checking the CSS I see that .modals.dimmer .ui.scrolling.position: relative!important; } is what is preventing the scrollbar from showing up.  However disabling it creates styling issues so not sure what to do..

**Comment by ktaras** [comment]
Just adding class `scrolling` doesn't help.
It fixes flickering, but when the content is large there is no scrollbar and the content just overflows through the modal borders. Any other suggestions? Maybe `scrolling` should be applied in complex with some other adjustments?

It happens after upgrading to 0.81 and later versions. 0.80.2 is ok.

**Comment by mgrivera** [comment]
Hi, I must say I'm very new to react & semantic-ui as well. 
So, version **0.80.2** is Ok, I assume for _semantic-ui-react_; I currently have version **0.86.0** and experiencing the issues mentioned here. 
My question is: what version should we use for _semantic-ui-css_; I currently am using version **2.4.0**. But what version should I use for _Modal_ to behave correctly? 
Thanks and bye ...

**Comment by layershifter** [comment]
I provided a fix in #3679, please try it, feedback there is much appreciated 🙌
